taskcluster/script.py

```diff
@@ -203,22 +261,32 @@ def main():
     rc = None
     bytes_read = 0
     bytes_written = 0
-    proc = subprocess.Popen(extra_args,
-                            bufsize=0,
-                            env=env,
-                            stdout=subprocess.PIPE,
-                            stderr=subprocess.STDOUT)
-    while True:
-        line = proc.stdout.readline()
-        decoded_line = line.decode()
-        line_len = len(decoded_line)
-        bytes_read += line_len
-        rc = proc.poll()
-        if line:
-            bytes_written += sys.stdout.write(decoded_line)
-        if line_len == 0 and bytes_written == bytes_read and rc is not None:
-            break
-    print("script.py: command finished (bytes read: %s, bytes written: %s)" % (bytes_read, bytes_written))
+    dpi = DebugPrinter(device)
+    # timeout in x minutes
+    with timeout(TIMEOUT_MINUTES * 60, dpi):
+        proc = subprocess.Popen(extra_args,
+                                bufsize=0,
+                                env=env,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.STDOUT)
+        dpi.print_to_logcat("command started")
+        while True:
+            line = proc.stdout.readline()
+            decoded_line = line.decode()
+            line_len = len(decoded_line)
+            bytes_read += line_len
+            rc = proc.poll()
+            if line:
+                temp_bytes_written = 0
+                while temp_bytes_written != line_len:
+                    temp_bytes_written += sys.stdout.write(decoded_line)
+                    if temp_bytes_written != line_len:
```

Comment:
ditto <=

Reply:
I think that this could hide the first underwrite if it the write sends it out before the print.

Done.